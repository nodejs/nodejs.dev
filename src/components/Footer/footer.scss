.footer {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin: 0;
  margin-top: auto;

  .footer__left {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    list-style: none;
    height: 100%;
  }

  .footer__right {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    list-style: none;
    height: 100%;
    margin-right: 20px;
  }

  .footer__left > li,
  .footer__right > li {
    font-size: 12px;
    margin-right: 32px;
    height: 24px;
  }

  .footer__left {
    li a {
      color: var(--color-text-primary);

      &:hover {
        color: var(--color-brand-primary);
      }
    }

    > li:first-of-type {
      padding-right: 12px;
      padding-top: 0;
      margin-right: 12px;
      border-right: 1px solid var(--black5);
    }
  }

  .footer__link {
    text-decoration: none;
  }
}

@media (max-width: 800px) {
  .footer {
    flex-direction: column;

    .footer__left {
      flex-direction: column;
      height: 10rem;
      justify-content: center;
      align-items: center;
      > li:first-of-type {
        padding-right: 0;
        margin-right: 0;
        border-right: none;
      }
    }

    .footer__right {
      margin: 0;

      > li:first-of-type {
        margin-right: auto;
        margin-left: 0;
      }

      > li {
        margin-left: 18px;
      }
    }

    .footer__left,
    .footer__right {
      padding: 6px 18px;
    }

    .footer__left > li,
    .footer__right > li {
      margin-right: 0;
      padding: 6px 0;
    }
  }
}
