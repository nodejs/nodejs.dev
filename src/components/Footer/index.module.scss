.footer {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0;
  margin-top: auto;

  @media (max-width: 800px) {
    flex-direction: column;
  }

  &__left {
    display: flex;
    flex-wrap: wrap;
    height: 100%;
    justify-content: space-between;
    list-style: none;

    li a {
      color: var(--color-text-primary);

      &:hover {
        color: var(--color-brand-primary);
      }
    }

    > li:first-of-type {
      border-right: 1px solid var(--black5);
      margin-right: 12px;
      padding-right: 12px;
      padding-top: 0;
    }

    @media (max-width: 800px) {
      align-items: center;
      flex-direction: column;
      height: 10rem;
      justify-content: center;

      > li:first-of-type {
        border-right: none;
        margin-right: 0;
        padding: 6px 0;
      }
    }
  }

  &__right {
    display: flex;
    flex-wrap: wrap;
    height: 100%;
    justify-content: space-between;
    list-style: none;
    margin-right: 20px;

    > li svg {
      height: 1.8rem;
      width: 1.8rem;
    }

    @media (max-width: 800px) {
      margin: 0;

      > li:first-of-type {
        margin-left: 0;
        margin-right: auto;
      }

      > li {
        margin-left: 18px;
      }
    }
  }

  &__left,
  &__right {
    > li {
      font-size: 12px;
      height: 24px;
      margin-right: 32px;

      @media (max-width: 800px) {
        margin-right: 0;
        padding: 6px 0;
      }
    }

    @media (max-width: 800px) {
      padding: 6px 18px;
    }
  }

  &__link {
    text-decoration: none;
  }
}
