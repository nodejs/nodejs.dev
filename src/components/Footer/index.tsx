import React from 'react';
import { FormattedMessage } from 'react-intl';
import { LocalizedLink as Link } from 'gatsby-theme-i18n';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faGithub,
  faSlack,
  faTwitter,
} from '@fortawesome/free-brands-svg-icons';
import styles from './index.module.scss';

const Footer = (): JSX.Element => (
  <footer className={styles.footer}>
    <ul className={styles.left}>
      <li>
        <a
          className={styles.link}
          target="_blank"
          href="https://trademark-policy.openjsf.org/"
          rel="noopener noreferrer"
          aria-label="Node.js Slack Link"
        >
          <FormattedMessage id="components.footer.links.trademark" />
        </a>
      </li>
      <li>
      <a 
          className={styles.link} 
          href="https://privacy-policy.openjsf.org/"
          >
          <FormattedMessage id="components.footer.links.privacy" />
        </a>
      </li>
      <li>
        <a
          className={styles.link}
          href="https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct"
        >
          <FormattedMessage id="components.footer.links.codeOfConduct" />
        </a>
      </li>
      <li>
        <Link className={styles.link} to="/about/security">
          <FormattedMessage id="components.footer.links.security" />
        </Link>
      </li>
      <li>
        <Link className={styles.link} to="/about">
          <FormattedMessage id="components.footer.links.about" />
        </Link>
      </li>
      <li>
        <Link className={styles.link} to="/blog">
          <FormattedMessage id="components.footer.links.blog" />
        </Link>
      </li>
    </ul>
    <ul className={styles.right}>
      <li>
        <FormattedMessage id="components.footer.links.openJsFoundation" />
      </li>
      <li>
        <a
          target="_blank"
          href="https://github.com/nodejs/node"
          rel="noopener noreferrer"
          aria-label="Node.js Github Page Link"
        >
          <span className="sr-only">
            <FormattedMessage id="components.footer.links.github" />
          </span>
          <FontAwesomeIcon
            icon={faGithub}
            color="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          target="_blank"
          href="https://twitter.com/nodejs"
          rel="noopener noreferrer"
          aria-label="Node.js Twitter Link"
        >
          <FontAwesomeIcon
            icon={faTwitter}
            color="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          target="_blank"
          href="https://slack.openjsf.org"
          rel="noopener noreferrer"
          aria-label="Node.js Slack Link"
        >
          <FontAwesomeIcon icon={faSlack} color="var(--color-text-secondary)" />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
