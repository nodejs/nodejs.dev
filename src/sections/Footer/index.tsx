import React from 'react';
import { FormattedMessage } from 'react-intl';
import { LocalizedLink as Link } from 'gatsby-theme-i18n';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faGithub,
  faSlack,
  faTwitter,
  faMastodon,
} from '@fortawesome/free-brands-svg-icons';
import styles from './index.module.scss';

const Footer = (): JSX.Element => (
  <footer className={styles.footer}>
    <ul className={styles.left}>
      <li>
        <a
          className={styles.link}
          target="_blank"
          href="https://trademark-policy.openjsf.org/"
          rel="noopener noreferrer"
          aria-label="Node.js Trademark Policy - PDF (Opens in new tab)"
        >
          <FormattedMessage id="components.footer.links.trademark" />
        </a>
      </li>
      <li>
        <a
          className={styles.link}
          target="_blank"
          href="https://privacy-policy.openjsf.org/"
          rel="noopener noreferrer"
          aria-label="Node.js Privacy Policy - PDF (Opens in new tab)"
        >
          <FormattedMessage id="components.footer.links.privacy" />
        </a>
      </li>
      <li>
        <a
          className={styles.link}
          href="https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct"
        >
          <FormattedMessage id="components.footer.links.codeOfConduct" />
        </a>
      </li>
      <li>
        <Link className={styles.link} to="/about/security">
          <FormattedMessage id="components.footer.links.security" />
        </Link>
      </li>
      <li>
        <Link className={styles.link} to="/about">
          <FormattedMessage id="components.footer.links.about" />
        </Link>
      </li>
      <li>
        <Link className={styles.link} to="/blog">
          <FormattedMessage id="components.footer.links.blog" />
        </Link>
      </li>
    </ul>
    <ul className={styles.right}>
      <li>
        <FormattedMessage id="components.footer.links.openJsFoundation" />
      </li>
      <li>
        <a
          target="_blank"
          href="https://github.com/nodejs/node"
          rel="noopener noreferrer"
          aria-label="Node.js Github Page (opens in new tab)"
        >
          <FontAwesomeIcon
            icon={faGithub}
            color="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          target="_blank"
          href="https://mastodon.social/@nodejs@social.lfx.dev"
          rel="noopener noreferrer"
          aria-label="Node.js Mastodon (opens in new tab)"
        >
          <FontAwesomeIcon
            icon={faMastodon}
            color="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          target="_blank"
          href="https://twitter.com/nodejs"
          rel="noopener noreferrer"
          aria-label="Node.js Twitter (opens in new tab)"
        >
          <FontAwesomeIcon 
          icon={faTwitter}
          color="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          target="_blank"
          href="https://slack.openjsf.org"
          rel="noopener noreferrer"
          aria-label="Node.js Slack (opens in new tab)"
        >
          <FontAwesomeIcon
          icon={faSlack}
          color="var(--color-text-secondary)"
          />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
