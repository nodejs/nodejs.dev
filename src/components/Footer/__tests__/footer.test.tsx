import React from 'react';
import { render } from '@testing-library/react';
import Footer from '..';

describe('Tests for Footer component', () => {
  it('renders correctly', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
