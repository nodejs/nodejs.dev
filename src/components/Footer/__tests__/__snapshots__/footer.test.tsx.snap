// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Footer component renders correctly 1`] = `
<div>
  <footer
    class="footer"
  >
    <ul
      class="footer__left"
    >
      <li>
        <a
          class="footer__link"
          href="/about/trademark"
        >
          Trademark Policy
        </a>
      </li>
      <li>
        <a
          class="footer__link"
          href="/about/privacy"
        >
          Privacy Policy
        </a>
      </li>
      <li>
        <a
          class="footer__link"
          href="https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct"
        >
          Code of Conduct
        </a>
      </li>
      <li>
        <a
          class="footer__link"
          href="/about/security"
        >
          Security Reporting
        </a>
      </li>
      <li>
        <a
          class="footer__link"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          class="footer__link"
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
    <ul
      class="footer__right"
    >
      <li>
        Â© OpenJS Foundation
      </li>
      <li>
        <a
          aria-label="Node.js Github Page Link"
          href="https://github.com/nodejs/node"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="sr-only"
          >
            GitHub
          </span>
          <svg
            fill="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          aria-label="Node.js Twitter Link"
          href="https://twitter.com/nodejs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="var(--color-text-secondary)"
          />
        </a>
      </li>
      <li>
        <a
          aria-label="Node.js Slack Link"
          href="https://slack.openjsf.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="var(--color-text-secondary)"
          />
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
